### PR TITLE
Revert "Revert "Provider > Add link API between on-prem VCS and Agent-Pool""

### PR DESCRIPTION
**Pull request by penja**
Reverts Scalr/terraform-provider-scalr#237

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `data.scalr_environment`: optional `id` and `name` arguments can be used together ([#228](https://github.com/Scalr/terraform-provider-scalr/pull/228))
 - `data.scalr_endpoint`: optional `id` and `name` arguments can be used together ([#228](https://github.com/Scalr/terraform-provider-scalr/pull/228))
 - `data.scalr_service_account`: optional `id` and `email` arguments can be used together ([#228](https://github.com/Scalr/terraform-provider-scalr/pull/228))
+- `scalr_vcs_provider`: added new attribute `agent_pool_id` ([#233](https://github.com/Scalr/terraform-provider-scalr/pull/233))
+- `data.scalr_vcs_provider`: added new attribute `agent_pool_id` ([#233](https://github.com/Scalr/terraform-provider-scalr/pull/233))
+- `scalr_agent_pool`: added new attribute `vcs_enabled` ([#233](https://github.com/Scalr/terraform-provider-scalr/pull/233))
+- `data.scalr_agent_pool`: added new attribute `vcs_enabled` ([#233](https://github.com/Scalr/terraform-provider-scalr/pull/232))
```
Comment (emocharnik) [inline CHANGELOG.md:29]: missing required server version

Comment (emocharnik) [reply CHANGELOG.md:29]: cc @kozliuk

**Comment by emocharnik** [comment]
@penja, Pull request has conflicts with base branch

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,7 +54,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - `scalr_account_allowed_ips`: accept /32 suffix ([#224](https://github.com/Scalr/terraform-provider-scalr/pull/224))
-- `data.scalr_module_version`: if there are several module versions with the same version, select the version that has the 'is-root-module' flag set to true. ([#229](https://github.com/Scalr/terraform-provider-scalr/pull/229))
```
Comment (petroprotsakh) [inline CHANGELOG.md:42]: I guess this was merged into [1.0.5] section by mistake after the v1.0.5 tag was created, so it's moving to [Unreleased] now, right?

**Review by petroprotsakh — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (kozliuk) [reply CHANGELOG.md:42]: yes, https://scalr-labs.atlassian.net/browse/SCALRCORE-25897

Comment (kozliuk) [reply CHANGELOG.md:42]: it will be release in 1.0.6

**Pull request merged**